judge/models/problem.py

```diff
@@ -451,6 +451,29 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def has_solved(self, user):
```

Comment:
I don't like this name. Is this a good name? What about `solved_by`? I feel like that makes the callsite more clear. Or maybe people don't care.

Reply:
changed to `solved_by`